app/derivative_services/scanned_map_derivative_service.rb

```diff
@@ -42,7 +42,19 @@ def parent
 
   def create_derivatives
     jp2_derivative_service.create_derivatives if jp2_derivative_service.valid?
-    thumbnail_derivative_service.create_derivatives if thumbnail_derivative_service.valid?
+    begin
+      thumbnail_derivative_service.create_derivatives if thumbnail_derivative_service.valid?
+    rescue StandardError => error
+      # failure_change_set = FileSetChangeSet.new(resource)
+      original_files = resource.file_metadata.reject(&:derivative?)
+      # failure_change_set.validate(file_metadata: original_files)
+      # change_set_persister.save(change_set: failure_change_set)
+
+      # Roll back the JP2000
+      resource.file_metadata = original_files
```

Comment:
This should delete the metadata for the JP2 derivative when the thumbnail generation fails, but I suspect that the file itself might also need to be explicitly deleted using the storage adapter.